libyul/optimiser/NameSimplifier.cpp

```diff
@@ -28,19 +30,13 @@
 using namespace solidity::yul;
 using namespace std;
 
-NameSimplifier::NameSimplifier(
-	OptimiserStepContext& _context,
-	Block const& _ast
-):
-	m_context(_context),
-	m_usedNames(_context.reservedIdentifiers)
+NameSimplifier::NameSimplifier(OptimiserStepContext& _context):
+	m_context(_context)
 {
-	for (YulString name: m_usedNames)
+	for (YulString name: _context.reservedIdentifiers)
 		m_translations[name] = name;
 
-	set<YulString> allNames = NameCollector(_ast).names();
-	m_usedNames += allNames;
-	for (YulString name: allNames)
+	for (YulString const& name: m_context.dispenser.usedNames())
```

Comment:
This assumes that the dispenser has been reset right before - I'm not sure we should make this assumption.

Reply:
This just assumes that Dispenser is initialized, there would be no issues if it was reset before or not.

Maybe it's just better to use `NameCollector(_ast).names()` it would make more sense. I'll fix it.

Reply:
Fixed this.